### PR TITLE
Update Tiny Tensor Compiler integration

### PR DESCRIPTION
**Pull request by uphoffc**
Updates plumbing for Tiny Tensor Compiler (tinytc).

* Add Abstract Syntax Tree for Tiny Tensor Language
* Refactor fused GEMM implementation to generate tinytc-AST instead of code
* Add copyscaleadd for tinytc (Loop-over-AXPBY)
* Make GemmForge optional for oneapi backend, tinytc serves as full replacement
* Update wrapper code to work with tinytc 0.3.1

### CODE REVIEW

**Comment by uphoffc** [comment]
@vikaskurapati @davschneller anybody got time to review this?

File: yateto/codegen/common.py [hunk 1]
```diff
@@ -122,3 +132,130 @@ def deduce_offset_arg(self, term):
...
+    """Kernel argument for TinytcWrapper.
+
+    Arguments:
+    name -- Argument name
+    call_expr -- Expression used in calling wrapper
+    constant -- Whether a tensor is invariant to group id
+    temporary -- Whether a tensor is stored in a temporary buffer
+    modified -- Whether tensor is modified during kernel
+    """
+    self.name = name
```
Comment (vikaskurapati) [inline yateto/codegen/common.py:148]: Just out of curiosity, why are there no underscores for class variables here but  used for TinytcWriter?

Comment (uphoffc) [reply yateto/codegen/common.py:148]: I typically add underscores to members considered private as Python does not have any private members. Let me know if you find any inconsistencies in the naming.

**Review by vikaskurapati — APPROVED** [review]
Seems fine to me as far as I understand. Thank you!

**Comment by davschneller** [comment]
Hi; just letting you know that we're planning to replace Yateto short-/mid-term by TensorForge—in case you were planning to change the structure of Yateto more fundamentally in near-future commits. It's probably going to happen once https://github.com/SeisSol/SeisSol/pull/1102 is ready. Right now, the front-end of the two repositories is still the same, but that might change pretty soon. We'd inform you once the internal structure is stabilized enough, so that you (or we) can add support for TinyTC there as well.

**Pull request merged**